rmf_traffic_editor/gui/editor.cpp

```diff
@@ -978,14 +979,41 @@ void Editor::keyPressEvent(QKeyEvent* e)
       }
       else
       {
-        QMessageBox::critical(
-          this,
-          "Could not delete item",
-          "If deleting a vertex, it must not be in any edges or polygons "
-          "or lifts. "
-          "If deleting a feature, it must not be in any constraints.");
+        DeleteDialog dialog(this);
+        if (dialog.exec() == QDialog::Accepted)
+        {
+          // Get the selected vertex index
+          int selected_vertex_idx = -1;
+          for (int i = 0;
+            i < static_cast<int>(building.levels[level_idx].vertices.size());
+            i++)
+          {
+            if (building.levels[level_idx].vertices[i].selected)
+            {
+              selected_vertex_idx = i;
+              break;
+            }
+          }
 
-        building.clear_selection(level_idx);
+          if (selected_vertex_idx != -1)
+          {
+            // Get the selected vertex
+            const auto v =
+              building.levels[level_idx].vertices[selected_vertex_idx];
+            auto it = v.params.find("lift_cabin");
+            if ((it != v.params.end()))
+            {
+              // Remove all vertices with a lift cabin property that matches the selected vertex
+              building.purge_lift_cabin_vertices(v.lift_cabin());
+            }
+          }
+
+          building.levels[level_idx].delete_used_entities();
+          undo_stack.push(new DeleteCommand(&building, level_idx));
+          create_scene();
+        }
+        else
+          building.clear_selection(level_idx);
```

Comment:
This is because the selection is not cleared properly.

This can be fixed by adding these 3 lines:
```
clear_current_tool_buffer();
update_property_editor();
create_scene();
```

This clears the tool selection, updates the property panel (clears it because the tool selection is cleared), and updates the scene to visually unselect the entity.

Reply:
Awesome!